spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -387,16 +407,36 @@ public void handleDelivery(String consumerTag, Envelope envelope,
 		}
 
 		@Override
+		public void handleConsumeOk(String consumerTag) {
+			super.handleConsumeOk(consumerTag);
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("New " + this);
+			}
+		}
+
+		@Override
 		public void handleCancelOk(String consumerTag) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("CancelOk " + this);
+			}
+			synchronized (DirectMessageListenerContainer.this.consumersMonitor) {
+				DirectMessageListenerContainer.this.cancellationLock.release(this);
+				DirectMessageListenerContainer.this.consumers.remove(this);
```

Comment:
Why don't we remove `Consumer` from the `consumersByQueue` as well, like we do that in the `handleCancel()`?